src/macros.h

```diff
@@ -143,6 +181,69 @@
 #define SIZE_OF(typ, id) sizeof(((typ *)0)->id)
 #endif // !SIZE_OF
 
+ /**
+ * In order to determine the offset of an element of a structure, use the
+ * following macro as shown:
+ * 
+ * Example usage:
+ *  typedef struct _ABC {
+ *      sint32    A;
+ *      sint32    B;
+ *      sint16    C;
+ * } ABC, * PTR_ABC
+ * 
+ * OFFSET_OF(struct _ABC, C)
+ */
+#ifndef OFFSET_OF
+#define OFFSET_OF(typ, id) (unsigned int)&(((typ*)0)->id)
+#endif // !OFFSET_OF
+
+ /**
+ * Determines the alignment of a type.
+ * 
+ * Example usage:
+ *  size_t alignment = ALIGN_OF(int);
+ */
+#ifndef ALIGN_OF
+#if defined(__MWERKS__) || defined(__CWCC__) // Metrowerks compilers exist for many platforms.
+#define ALIGN_OF(type) ((size_t)__alignof__(type))
+#elif !defined(COMPILER_GNUC) || (__GNUC__ >= 3) // GCC 2.x doesn't do __alignof correctly all the time.
+#define ALIGN_OF __alignof
+#else
+#define ALIGN_OF(type) ((size_t)OFFSET_OF(struct{ char c; type m; }, m))
+#endif // Metrowerks
+#endif // !ALIGN_OF
+
+/**
+ * Safely delete an object and set the pointer to NULL.
+ * 
+ * Example usage:
+ *  SAFE_DELETE(MyPtr);
+ */
+#ifndef SAFE_DELETE
+#define SAFE_DELETE(ptr) if ((ptr) != nullptr) { delete (ptr); } (ptr) = nullptr;
+#endif // !SAFE_DELETE
+
+/**
+ * Safely delete an array and set the pointer to NULL.
+ * 
+ * Example usage:
+ *  SAFE_DELETE_ARRAY(MyArray);
+ */
+#ifndef SAFE_DELETE_ARRAY
+#define SAFE_DELETE_ARRAY(ptr) if ((ptr) != nullptr) { delete [] (ptr); } (ptr) = nullptr;
+#endif // !SAFE_DELETE_ARRAY
+
+/**
+ * Mark variables as unused.
+ * 
+ * Example usage:
+ *  UNUSED_VAR(MyBool);
+ */
+#ifndef UNUSED_VAR
```

Comment:
Not really useful in my opinion, just cast to void if you get a warning about this, remove the var if it really isn't used or disable the warning for the project.